manifests/value.pp

```diff
@@ -57,7 +57,8 @@
   validate_re($type, '^\w+',
     'type parameter must not be empty but it is type => "$type"')
 
-
+  # normalize the case so that our defined() check is more consistent
+  $key_normalized = downcase($key)
```

Comment:
In that case, they can use a `registry_value` resource directly, I guess? IMHO, normalizing to a known case is less of a surprise than arbitrarily breaking when they use inconsistent casing, since they're used to having that freedom.